internal/command/instance.go

```diff
@@ -959,18 +959,23 @@ func getSystemConfigWriteModel(ctx context.Context, filter preparation.FilterToQ
 	return writeModel, err
 }
 
-func (c *Commands) RemoveInstance(ctx context.Context, id string) (*domain.ObjectDetails, error) {
+func (c *Commands) RemoveInstance(ctx context.Context, id string, errorIfNotFound bool) (*domain.ObjectDetails, error) {
 	instID := strings.TrimSpace(id)
 	if instID == "" || len(instID) > 200 {
 		return nil, zerrors.ThrowInvalidArgument(nil, "COMMA-VeS2zI", "Errors.Invalid.Argument")
 	}
 
 	instanceAgg := instance.NewAggregate(instID)
-	cmds, err := preparation.PrepareCommands(ctx, c.eventstore.Filter, c.prepareRemoveInstance(instanceAgg))
+	//nolint: staticcheck
+	cmds, err := preparation.PrepareCommands(ctx, c.eventstore.Filter, c.prepareRemoveInstance(instanceAgg, errorIfNotFound))
 	if err != nil {
 		return nil, err
 	}
 
+	if len(cmds) == 0 {
+		return nil, nil
```

Comment:
If you use the writeModelToObjectDetails function you can return the current state of the instance aggregate.